python/revng/cli/_commands/ptml.py

```diff
@@ -0,0 +1,273 @@
+
+    def register_arguments(self, parser: argparse.ArgumentParser):
+        subparsers = parser.add_subparsers(title="Subcommands", metavar="")
+
+        parser_cat = subparsers.add_parser("cat", help="Print the ptml on the console")
+        parser_cat_filter = parser_cat.add_mutually_exclusive_group()
+        parser_cat_filter.add_argument(
+            "-f",
+            "--filter",
+            type=str,
+            action="append",
+            default=[],
+            required=False,
+            help="Only show the keys specified (if present)",
+        )
+        parser_cat_filter.add_argument(
+            "-e", "--extract", type=str, required=False, help="Extract the specified key"
+        )
+        parser_cat_color = parser_cat.add_mutually_exclusive_group()
+        parser_cat_color.add_argument(
+            "-p", "--plain", action="store_true", help="Force plaintext output"
+        )
+        parser_cat_color.add_argument(
+            "-c", "--color", action="store_true", help="Force color output"
+        )
+        parser_cat.add_argument(
+            "input",
+            type=argparse.FileType("r"),
+            default=sys.stdin,
+            nargs="?",
+            help="Input file (stdin if omitted)",
+        )
+        parser_cat.set_defaults(func=cmd_cat)
+
+        parser_strip = subparsers.add_parser("strip", help="Strip PTML from a file")
+        parser_strip_filter = parser_strip.add_mutually_exclusive_group()
+        parser_strip_filter.add_argument(
+            "-f",
+            "--filter",
+            type=str,
+            action="append",
+            default=[],
+            required=False,
+            help="Only show the keys specified (if present)",
+        )
+        parser_strip_filter.add_argument(
+            "-e", "--extract", type=str, required=False, help="Extract the specified key"
+        )
+        parser_strip.add_argument(
+            "input",
+            type=argparse.FileType("r+"),
+            default=sys.stdin,
+            nargs="?",
+            help="Input file (stdin if omitted)",
+        )
+        parser_strip_out = parser_strip.add_mutually_exclusive_group()
+        parser_strip_out.add_argument("-i", "--inplace", action="store_true", help="Strip inplace")
+        parser_strip_out.add_argument(
+            "output",
+            type=argparse.FileType("w"),
+            default=sys.stdout,
+            nargs="?",
+            help="Output file (stdout if omitted)",
+        )
+        parser_strip.set_defaults(func=cmd_strip)
+
+    def run(self, options: Options):
+        if "func" not in options.parsed_args:
+            sys.exit(1)
+
+        def new_unraisablehook(arg):
+            if arg.exc_type != BrokenPipeError:
+                sys.__unraisablehook__(arg)
+
+        sys.unraisablehook = new_unraisablehook
+
+        with suppress(BrokenPipeError, KeyboardInterrupt):
+            options.parsed_args.func(options.parsed_args)
```

Comment:
This stuff is suspicious, document.

Reply:
Added docstring and isolated as `suppress_brokenpipe`